src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,405 @@
+  smart_objects::SmartArray& ar_cmd = *(commands.asArray());
+  ar_vr_cmd.reserve(count_vr_commands_ + count_to_add);
+  ar_vr_cmd.insert(
+      ar_vr_cmd.end(), ar_cmd.begin(), ar_cmd.begin() + count_to_add);
+
+  LOG4CXX_DEBUG(logger_,
+                "VR commands with id: " << cmd_id << " added for appID: "
+                                        << app_.app_id());
+  count_vr_commands_ += count_to_add;
+  return true;
+}
+
+bool HelpPromptManagerImpl::DeleteCommand(const uint32_t cmd_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(vr_commands_lock_);
+
+  auto it = vr_commands_.find(cmd_id);
+  if (vr_commands_.end() == it) {
+    LOG4CXX_WARN(logger_, "VR command with id: " << cmd_id << " not found");
+    return false;
+  }
+
+  count_vr_commands_ -= it->second->length();
+
+  vr_commands_.erase(it);
+  LOG4CXX_DEBUG(
+      logger_,
+      "VR command with id: "
+          << cmd_id << " deleted for appID: " << app_.app_id()
+          << ". Remaining number of commands: " << count_vr_commands_);
+  return true;
+}
+
+void HelpPromptManagerImpl::OnVrCommandAdded(
+    const uint32_t cmd_id,
+    const smart_objects::SmartObject& command,
+    const bool is_resumption) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (SendingType::kNoneSend == sending_type_) {
+    LOG4CXX_DEBUG(logger_,
+                  "SendingType::kNoneSend"
+                      << " commands with id:" << cmd_id
+                      << " will not be added");
+    return;
+  }
+  if (AddCommand(cmd_id, command) && !is_resumption) {
+    SendRequests();
+  }
+}
+
+void HelpPromptManagerImpl::OnVrCommandDeleted(const uint32_t cmd_id,
+                                               const bool is_resumption) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (SendingType::kNoneSend == sending_type_) {
+    LOG4CXX_DEBUG(logger_,
+                  "SendingType::kNoneSend"
+                      << " commands with id:" << cmd_id
+                      << " will not be deleted");
+    return;
+  }
+  if (DeleteCommand(cmd_id) && !is_resumption) {
+    SendRequests();
+  }
+}
+
+void HelpPromptManagerImpl::OnSetGlobalPropertiesReceived(
+    const smart_objects::SmartObject& msg, const bool is_response) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (SendingType::kNoneSend == sending_type_) {
+    LOG4CXX_DEBUG(logger_,
+                  "SendingType::kNoneSend"
+                  " do not track SetGlobalProperties");
+    return;
+  }
+
+  if (!is_response) {
+    if (msg.keyExists(strings::help_prompt)) {
+      is_tts_send_ = true;
+    }
+    if (msg.keyExists(strings::vr_help)) {
+      is_ui_send_ = true;
+    }
+
+    LOG4CXX_DEBUG(logger_, "is_tts_send_:" << is_tts_send_);
+    LOG4CXX_DEBUG(logger_, "is_ui_send_:" << is_ui_send_);
+    return;
+  }
+
+  SetSendingType(msg);
+}
+
+HelpPromptManagerImpl::SendingType HelpPromptManagerImpl::GetSendingType()
+    const {
+  if (is_tts_send_ && is_ui_send_) {
```

Comment:
This was never fixed, the body of this function should just be `return sending_type_;`

Reply:
@jacobkeeler sorry, didn't get your comment. Fixed in 1de1700